### PR TITLE
Added algorithm for Group Anagram

### PR DESCRIPTION
**Pull request by shreyasharma6602**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains
 
Fixes: #6527 .

### CODE REVIEW

**Comment by sofezticated-gentleman** [comment]
hey @shreyasharma6602 I am facing the issue of requires doctest can you please help me out ? what is doctest?

**Comment by shreyasharma6602** [comment]
> hey @shreyasharma6602 I am facing the issue of requires doctest can you please help me out ? what is doctest?

Doctest is a module which is used for better testing purpose. You can try to implement it from these links.
[Doctest](https://www.geeksforgeeks.org/testing-in-python-using-doctest-module/)
[Doctest Python Documentation](https://docs.python.org/3/library/doctest.html)

**Comment by sofezticated-gentleman** [comment]
thanks @shreyasharma6602

**Comment by cclauss** [comment]
What a terrible title for a pull request.  The title should explain what the PR does and the issue number should be in the commit message so the reader can click on it.

File: strings/group_anagrams.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+from typing import List
```
Comment (cclauss) [inline strings/group_anagrams.py:1]: We only support the latest version of CPython so you can replace all occurrences of `List` with `list`

File: strings/group_anagrams.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+def group_anagrams(words: List[str]) -> dict:
+  """
+  >>> group_anagrams(['cloud', 'python', 'could'])
+  {'cdlou': ['cloud', 'could'], 'hnopty': ['python']}
+  >>> group_anagrams(['strap', 'traps', 'maps'])
+  {'aprst': ['strap', 'traps'], 'amps': ['maps']}
+  >>> group_anagrams(['astro', 'plastic', 'kite'])
+  {'aorst': ['astro'], 'acilpst': ['plastic'], 'eikt': ['kite']}
+  """
+  anagrams = dict()
```
Comment (cclauss) [inline strings/group_anagrams.py:24]: How could this code be tighter by using:
```suggestion
  anagrams = defaultdict(list)
```
https://docs.python.org/3/library/collections.html#collections.defaultdict

File: strings/group_anagrams.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
+from typing import List
+import doctest
```
Comment (cclauss) [inline strings/group_anagrams.py:2]: Let's move this line down under `if __name__ == "__main__":` so it is not imported if someone uses this file as a module.

File: strings/group_anagrams.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+def signature(word: str) -> str:
+  """
```
Comment (cclauss) [inline strings/group_anagrams.py:5]: Four space indents like PEP8 recommends.

**Comment by cclauss** [comment]
Please read CONTRIBUTING.md.

Hacktoberfest is strenuous for our team of active volunteer maintainers. That small team now has ___200+ open pull requests___ to review.  We have __high standards for our submissions__ which are described in [`CONTRIBUTING.md`](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md#contributing-guidelines).  Our continuous integration tests help to ensure that our standards are met before we review and merge deserving pull requests.  This is further complicated by the fact that we already have [lots of algorithms](https://github.com/TheAlgorithms/Python/blob/master/DIRECTORY.md) and would like to avoid duplicates.

For these reasons, I will **close all open pull requests failing tests** so that our unpaid team has a smaller mountain to climb.  If you want us to land your pull request, please read [`CONTRIBUTING.md`](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md#contributing-guidelines) _carefully_ and try again.

**Pull request closed without merging**